_posts/en/newsletters/2025-02-07-newsletter.md

```diff
@@ -0,0 +1,533 @@
+  changes:
+
+  - [Introduce legacy input sigops limit][ap1]: in a private thread,
+    Poinsot and several other contributors have attempted to produce a
+    block for regtest that will take the longest possible time to
+    validate using known problems in the validation of legacy
+    (pre-segwit) transactions.  After research, he found that he could
+    "adapt [the worst block] to be valid under the [mitigations
+    originally proposed][ccbip] in 2019" (see [Newsletter #36][news36
+    cc]).  This leads him to propose a different mitigation: limiting
+    the maximum number of signature operations (sigops) in legacy
+    transactions to 2,500.  Each execution of `OP_CHECKSIG` counts as
+    one sigop and each execution of `OP_CHECKMULTISIG` can count as up
+    to 20 sigops (depending on the number of public keys used with it).
+    His analysis shows that this will decrease the worst-case validation
+    time by 97.5%.
+
+    As with any change of this type, there is a risk of [accidental
+    confiscation][topic accidental confiscation] due to the new rule
+    making previously signed transactions invalid.  If you know of
+    someone who requires legacy transactions with more than 2,500
+    single-sig operations or more than 2,125 keys for multisig
+    operations[^2kmultisig], please alert Poinsot or other protocol
+    developers.
+
+  - [Increase timewarp grace period to 2 hours][ap2]: previously, the
+    cleanup proposal disallowed the first block in a new difficulty
+    period from having a block header time more than 600 seconds before
+    the time of the previous block.  This meant that a constant amount
+    of hashrate could not use the [timewarp][topic time warp]
+    vulnerability to produce blocks faster than once every 10 minutes.
+
+    Poinsot now accepts using a 7,200 second (2 hour) grace period, as
+    originally proposed by Sjors Provoost as being much less likely to
+    lead to a miner accidentally producing an invalid block, although it
+    does allow a very patient attacker controlling more than 50% of
+    network hashrate can use the timewarp attack to lower difficulty
+    over a period of months even if actual hashrate remains constant or
+    increases.  This would be a publicly visible attack and the network
+    would have months to react.  In his post, Poinsot summarizes
+    previous arguments (see [Newsletter #335][news335 cc] for our much
+    less detailed summary) and concludes that, "despite the fairly weak
+    arguments in favor of increasing the grace period, the cost of doing
+    so [does] not prohibit erring on the safe side."
+
+    On a thread dedicated to discussing extending the grace period,
+    developers Zazy and Pieter Wuille [discussed][wuille erlang] how the
+    600 second grace period, which would seem to allow a small amount of
+    time warping, was actually sufficient to ensure constant difficulty
+    under constant hashrate.  Specifically, they looked at the impact of
```

Comment:
Under constant hashrate **and** under attack. Pieter's point is that the 600 grace period makes it so the worst case under an attack only increases the block rate to what it was originally advertized to be (1 block per 10 minutes on average).